code/core/src/preview-api/modules/store/GlobalsStore.ts

```diff
@@ -62,5 +62,11 @@ export class GlobalsStore {
 
   update(newGlobals: Globals) {
     this.globals = { ...this.globals, ...this.filterAllowedGlobals(newGlobals) };
+
+    for (const key in newGlobals) {
+      if (newGlobals[key] === undefined) {
+        this.globals[key] = this.initialGlobals[key];
```

Comment:
Deleting wasn't doing the trick. We have to reset it to the initial "initialGlobals" value instead. That's what the comparing logic later requires.